hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterJoinManager.java

```diff
@@ -685,6 +704,24 @@ private boolean checkIfJoinRequestFromAnExistingMember(JoinMessage joinMessage,
         return true;
     }
 
+    private boolean isMemberRestartingWithPersistence(Member member) {
+        return member.getAttributes().get(PERSISTENCE_ENABLED_ATTRIBUTE) != null
+                && member.getAttributes().get(PERSISTENCE_ENABLED_ATTRIBUTE).equals("true")
+                // may be already detected as crashed member or probably it is still in member list because
+                // connection timeout hasn't been reached yet
+                && (hasMemberLeft(member.getUuid())
+                    || clusterService.getMembershipManager().getMember(member.getUuid()) != null);
```

Comment:
was the following intended?
```suggestion
                    || clusterService.getMembershipManager().getMember(member.getUuid()) == null);
```

Reply:
Yes, it is intended. In order for member to be found in members-left table, it must have already been removed from members list. This happens after connection error is detected and configured reconnection attempts fail/timeout. If the member rejoins while its departure from cluster hasn't been applied yet to the members list, we still want to allow it to join and perform the recovery process.

Reply:
Ohh, I see, makes sense, thanks